tests/cli.rs

```diff
@@ -213,14 +212,21 @@ fn test_write_all_values(#[case] number: i32, #[case] expected_output: &str) ->
     let command_assert = cmd.assert();
     let output = command_assert.get_output();
     let stdout = String::from_utf8_lossy(&output.stdout);
-    assert_eq!(stdout.trim_end(), expected_output);
+    assert_eq!(
+        Element::read_all(stdout.trim_end())?,
+        Element::read_all(expected_output)?
+    );
     Ok(())
 }
 
 #[cfg(feature = "experimental-code-gen")]
-#[rstest]
-#[case::simple_struct(
-    r#"
+mod code_gen_tests {
```

Comment:
🗺️ All of the changes in this file come from introducing a new module that could be more easily feature gated behind `experimental-code-gen`. There are no logic changes.